src/toast/tests/ops_filterbin.py

```diff
@@ -616,3 +617,156 @@ def test_filterbin_obsmatrix_cached(self):
+                input_map = hp.reorder(input_map, r2n=True)
+            hp.write_map(input_map_file, input_map, nest=pixels.nest, column_units="K")
+
+        # Scan map into timestreams
+        scan_hpix = ops.ScanHealpixMap(
+            file=input_map_file,
+            det_data=defaults.det_data,
+            pixel_pointing=pixels,
+            stokes_weights=weights,
+        )
+        scan_hpix.apply(data)
+
+        # Copy the signal
+        ops.Copy(detdata=[(defaults.det_data, "signal_copy")]).apply(data)
+
+        # Configure and apply the filterbin operator
+        binning = ops.BinMap(
+            pixel_dist="pixel_dist",
+            covariance="covariance",
+            det_data=defaults.det_data,
+            pixel_pointing=pixels,
+            stokes_weights=weights,
+            noise_model=default_model.noise_model,
+            sync_type="allreduce",
+            shared_flags=defaults.shared_flags,
+            shared_flag_mask=detpointing.shared_flag_mask,
+            det_flags=defaults.det_flags,
+            det_flag_mask=255,
+        )
+
+        filterbin = ops.FilterBin(
+            name="filterbin",
+            det_data=defaults.det_data,
+            det_flags=defaults.det_flags,
+            det_flag_mask=255,
+            shared_flags=defaults.shared_flags,
+            shared_flag_mask=detpointing.shared_flag_mask,
+            binning=binning,
+            ground_filter_order=5,
+            split_ground_template=True,
+            poly_filter_order=2,
+            output_dir=self.outdir,
+            write_invcov=True,
+            write_obs_matrix=True,
+            poly_filter_view="scanning",
+        )
+
+        # Build the observation matrix twice, first in a single run and
+        # then running each observation separately and saving the
+        # noise-weighted matrix.
+
+        filterbin.name = "split_run"
+        filterbin.apply(data)
+
+        filterbin.name = "noiseweighted_run"
+        filterbin.write_invcov = True
+        filterbin.det_data = "signal_copy"
+        filterbin.noiseweight_obs_matrix = True
+
+        orig_name_filterbin = filterbin.name
+        orig_comm = data.comm
+        new_comm = Comm(world=data.comm.comm_group)
+
+        for iobs, obs in enumerate(data.obs):
+            # Data object that only covers one observation
+            obs_data = data.select(obs_uid=obs.uid)
+            # Replace comm_world with the group communicator
+            obs_data._comm = new_comm
+            filterbin.reset_pix_dist = True
+            filterbin.name = f"{orig_name_filterbin}_{obs.name}"
+            filterbin.apply(obs_data)
+            close_data(obs_data)
+
+        if data.comm.world_rank == 0:
+            import matplotlib.pyplot as plt
+
+            # Assemble the single-run matrix
+
+            rootname = os.path.join(self.outdir, f"split_run_obs_matrix")
+            #try:
+            fname_matrix = ops.combine_observation_matrix(rootname)
```

Comment:
I don't see where this is defined in the `toast.ops` module.  Maybe a typo or a file is not committed?

Reply:
Sorry, completely missed that import in `toast.ops.__init__.py`!